### DISCUSSION TITLE
'NoneType' object has no attribute 'loadMolecule'

### DISCUSSION THREAD

**Discussion by niuzhenjiang** [epam/ketcher#1253] [Q&A] [answered]
I deployed the remote version of ketcher, the version is 2.4.1. The indigo service docker version is deployed normally, and /v2/info can obtain data, but v2/indigo/check or v2/indigo/convert will report an error: 'NoneType' object has no attribute 'loadMolecule'. After checking, I found that the submitted form The value in the struct is incorrect and should be in mol format. How should this need to be changed? Thank you so much.
![image](https://user-images.githubusercontent.com/42111602/153702891-1f3a60df-5082-4664-9be9-51e10a7ba989.png)
![image](https://user-images.githubusercontent.com/42111602/153702910-5116f797-f05a-4a57-ba09-1d1ce83ce21e.png)
![image](https://user-images.githubusercontent.com/42111602/153702921-9b0a2e06-f77d-4a58-9096-caf3b7408415.png)
![image](https://user-images.githubusercontent.com/42111602/153702941-b48ef60b-b6c5-4265-ae8d-b09d8e81bc66.png)

**Comment by niuzhenjiang** [comment]
The log I received has a record in it, but I don't know how to modify it to make it work, thanks a lot.
![image](https://user-images.githubusercontent.com/42111602/153712486-ce3fd10b-7400-4011-9627-2f3e4720515b.png)

**Comment by niuzhenjiang** [comment]
After manually setting the struct to mol format, the request works fine.
![image](https://user-images.githubusercontent.com/42111602/153713675-8c6c5078-2abf-47f5-84fe-d2ce72a8cea9.png)

**Comment by ElenaOdnoshivkina** [comment]
Hi @niuzhenjiang 
We thank you for your interest in the Ketcher. Please specify which version of Indigo you are using?
We look forward to your reply so that we can resolve this issue as soon as possible.

**Comment by niuzhenjiang** [comment]
@ElenaOdnoshivkina 
hello,I am using the latest version of indigo service![image](https://user-images.githubusercontent.com/42111602/153838231-c7b1fdc0-b992-4196-8b56-6ebcf3b1fde3.png)

**Comment by niuzhenjiang** [comment]
I tested it myself, and the latest version of indigo service will prompt that there is no aromatize-skip-superatoms attribute in options.
![image](https://user-images.githubusercontent.com/42111602/153838693-cec726b9-b2bd-400f-87f7-72c6b46564ea.png)
and when the front-end request is submitted, the format is in ket format. I change them to mol format and delete the attribute aromatize-skip-superatoms, and then I can request normally.
![image](https://user-images.githubusercontent.com/42111602/153839619-42c4a1ed-a1c7-45c5-abf1-e2b5de2e881c.png)

However, there are still some problems. For example, an error will still be reported when requesting the convert interface. Maybe the changes I made are not very comprehensive, and there is no file downloaded when requesting to download a picture, but the picture data is received in the viewing network, but Whether it is in png format or svg format, all the returned data is in svg format.

**Comment by ElenaOdnoshivkina** [comment]
@niuzhenjiang 
You can check which version of Indigo you are using in the response to the /info request.
![image](https://user-images.githubusercontent.com/89905163/153839897-2d73fd11-2455-4444-aa60-fa663fa8c362.png)

**Comment by niuzhenjiang** [comment]
@ElenaOdnoshivkina 
![image](https://user-images.githubusercontent.com/42111602/153840367-58d0b20f-4b3d-4143-8325-5bf45ece1ba2.png)

**Comment by AndreiMazol** [comment, marked as answer]
Dear @niuzhenjiang ,

starting from ketcher v2.4.x, internal KET format is used for interaction with server side in order to  support all new features such as simple objects, text objects and enhanced stereochemistry (mol v2000 does not support these features). So it is necessary to  use indigo v1.6.x.  I will contact with Indigo team to get a link  to the latest indigo image and will share it in the next message in this thread. Sorry for the inconvenience.

Regards,
Andrei

**Comment by niuzhenjiang** [comment]
@AndreiMazol 
That's it, then I'll try to use indigo 1.6.x to test it. Thanks again, ketcher is a very good editor!

**Comment by even1024** [comment]
@niuzhenjiang , here is the link with docker hub tags: https://hub.docker.com/r/epmlsop/indigo-service/tags

**Comment by niuzhenjiang** [comment]
@even1024 
thanks a lot!

**Comment by niuzhenjiang** [comment]
@even1024 @AndreiMazol 
sorry, i want ask docker start command for indigo-service1.6.x:
```
docker run --restart=always -d -p 8002:8002 -e "INDIGO_UWSGI_RUN_PARAMETERS=--plugin python3 --py-autoreload=1"
-e "PYTHONPATH=/srv/indigo-python" -e "PYTHONDONTWRITEBYTECODE=1" --name=indigo_service
epmlsop/indigo-service:latest /bin/sh -c "supervisord -n"
```
This command doesn't seem to work, the container keeps restarting and can't run properly

**Comment by niuzhenjiang** [comment]
The docker command I am using now is:
```
docker run --restart=always -d -p 80:80 --name=indigo_service epmlsop/indigo-service:latest /bin/sh -c "gunicorn --bind 0.0.0.0:80 --workers=5 app:app"
```
I don't know if this is possible, please correct me if I'm wrong, thanks again!
